app/services/assessoravailability/AssessorService.scala

```diff
@@ -58,29 +58,29 @@ trait AssessorService {
     }
   }
 
-  def addAvailability(userId: String, assessorAvailability: model.exchange.AssessorAvailabilityOld): Future[Unit] = {
+  def addAvailability(userId: String, assessorAvailabilities: List[model.exchange.AssessorAvailability]): Future[Unit] = {
+    def toPersisted(availabilities: List[model.exchange.AssessorAvailability]) =
+      availabilities.map { exchange => model.persisted.assessor.AssessorAvailability.apply(exchange) }
+
     assessorRepository.find(userId).flatMap {
       case Some(existing) =>
-        val newAvailability = assessorAvailability.availability.flatMap { case (k, dates) =>
-          dates.map { date =>
-            model.persisted.assessor.AssessorAvailability(k, date)
-          }
-        }.toList
-        val mergedAvailability = existing.availability ++ newAvailability
-
-        val assessorAvailabilityToPersist = assessor.Assessor(userId, existing.skills, existing.civilServant, mergedAvailability,
+        val newAvailabilities = toPersisted(assessorAvailabilities)
```

Comment:
We do not want to expose the implicit conversion to the rest of the application.